atlp-rwanda/script-heroes-frontend
# #174797464 Reset Password

#### What does this PR do?

This PR implements the whole process for resetting the password. 

#### Description of Task to be completed?

- create forgot password UI.
- Create reset password UI.
- Validate both UI
- Create an action for forgot password.
- Create a reducer for forgot password.
- Create a reducer for reset password.

#### How should this be manually tested?

- Clone this repo `https://github.com/atlp-rwanda/script-heroes-frontend.git`.
- Check out this branch `ft-reset-password-#174797464`.
- Run `yarn`.
- Run `yarn start`.
- Register your email.
- Open this endpoint `/forgot`.
- Enter your email in the input box.
- Submit form
- Check if there is an email with reset password sent to you
- Click the reset button, it should redirect you to the reset password form.
- Fill in the form and submit.
- Try to login with the new password.
#### What are the relevant pivotal tracker stories?

[#174797464](https://www.pivotaltracker.com/story/show/174797464)
#### Screenshots (if appropriate)
<img width="1440" alt="Screen Shot 2020-12-14 at 16 57 07" src="https://user-images.githubusercontent.com/68109866/102097571-c46fa180-3e2e-11eb-9053-50a2261674c0.png">
<img width="1440" alt="Screen Shot 2020-12-14 at 17 07 30" src="https://user-images.githubusercontent.com/68109866/102097707-ec5f0500-3e2e-11eb-9ffd-f17ac821693a.png">
<img width="1440" alt="Screen Shot 2020-12-14 at 17 07 42" src="https://user-images.githubusercontent.com/68109866/102097729-f1bc4f80-3e2e-11eb-9fb1-96475924d244.png">
